configs/ci/run-solo.yaml

```diff
@@ -0,0 +1,9 @@
+plugin_group:
```

Comment:
I am not sure I understand the intent of this file. With the exception of `audio` and `offline_imu_cam`, all the plugins here are dependent on other plugins, and running them solo will not really test anything. And given that both `audio` and `offline_imu_cam` are going to be a part of a flow, why is this file required at all?

Reply:
I would argue that our plugin set and our Switchboard implementation should be well behaved enough that a plugin running in isolation should not affect the stability of the system. While I don't know semantically what this should mean for pure Phonebook services yet, I think plugins that are producers and consumers over Switchboard should launch and drain without crashing at a minimum.

There are also edge cases about the system design that might be covered by running plugins in isolation. For example, have we ever seen a case in the wild where Switchboard's queues fill up because there are no available consumers? We can extend our `run-solo` Ii to generate new test case scenarios.

We can also [see `run-solo` at work](https://github.com/ILLIXR/ILLIXR/runs/2564890077?check_suite_focus=true#step:10:2633) now that we have both Ubuntu 18 and 20 in our CI:
- Ubuntu 18 crashes on the `offline_imu_cam` plugin, with a curious mention to a VTK missing reference on drain (`dlclose`).
- Ubuntu 20 makes it past the `run-solo` section, and crashes during the [`pose_lookup` flow](https://github.com/ILLIXR/ILLIXR/runs/2564889969?check_suite_focus=true#step:10:2840).

Reply:
Gotcha. Just so we're on the same page: the idea here is to make sure that the plugin actually doesn't do anything, like it's supposed to (if there are no pubs)?

Reply:
> Gotcha. Just so we're on the same page: the idea here is to make sure that the plugin actually doesn't do anything, like it's supposed to (if there are no pubs)?

Right! For now I think that's sufficient. If the plugin interface changes of course, we can reconsider what the behavior should be.

Reply:
In terms of execution time, I think we can play with the `RUN_DURATION` here if we find CI takes too long in the `run-solo` section. The `run-solo` section is also parallelizable if needed, since ILLIXR with just one plugin will not need many resources.

Parallelizing flows is trickier, since each flow would be a full ILLIXR app running on the CI machine. Ideally, we would launch each flow on a dedicated CI machine, but this requires exposing more functionality via Docker+Runner to GitHub Actions I think.

Reply:
> Gotcha. Just so we're on the same page: the idea here is to make sure that the plugin actually doesn't do anything, like it's supposed to (if there are no pubs)?

Plugins that depend on HW can perform their detection too, if we are able to passthrough the right resources via virtualization.